### PR TITLE
Chore: Make Cfg field private in SQLStore

### PR DESCRIPTION
**Pull request by zserge**
A follow-up to https://github.com/grafana/grafana/pull/85366 - it turns Cfg (`*setting.Cfg`) field into a private one inside SQLStore (and also hides `Dialect` field, as a bonus). This makes SQLStore functionally equivalent to `db.DB` interface.

### CODE REVIEW

File: pkg/infra/db/db.go [hunk 1]
```diff
@@ -59,8 +59,7 @@ var CleanupTestDB = sqlstore.CleanupTestDB
 var ProvideService = sqlstore.ProvideService
 
 func InitTestDBwithCfg(t sqlutil.ITestDB, opts ...InitTestDBOpt) (*sqlstore.SQLStore, *setting.Cfg) {
-	store := InitTestDB(t, opts...)
-	return store, store.Cfg
+	return InitTestDB(t, opts...)
```
Comment (DanCech) [inline pkg/infra/db/db.go:62]: Should we just replace InitTestDBwithCfg altogether across the codebase?

Comment (zserge) [reply pkg/infra/db/db.go:62]: Yep, changing it all over, so the final PR would be much bigger

**Comment by DanCech** [comment]
I'm confused, now we have `sqlstore.InitTestDB()` that returns sqlstore and config, then `db.InitTestDB()` that only returns sqlstore, and `db.InitTestDBWithCfg()` that returns both?

I was suggesting that `db.InitTestDB()` should return both (same as `sqlstore.InitTestDB()`), then we could switch anywhere using `db.InitTestDBWithCfg` to just say `sqlstore, _ = db.InitTestDB(t)` and remove InitTestDBWithCfg altogether.

**Review by marefr — APPROVED** [review]
LGTM 🎉 Based on explanation sounds like you have a migration path moving forward to clear up any confusion and this is a needed intermediate step 👍

**Review by DanCech — APPROVED** [review]

**Review by ryantxu — APPROVED** [review]
🚀

**Review by DanCech — APPROVED** [review]

**Pull request merged**